hugr-core/src/extension/resolution/types.rs

```diff
@@ -0,0 +1,176 @@
+//! Resolve weak links inside `CustomType`s in an optype's signature.
+
+use std::sync::Arc;
+
+use super::{ExtensionRegistry, ExtensionResolutionError};
+use crate::ops::OpType;
+use crate::types::type_row::TypeRowBase;
+use crate::types::{CustomType, MaybeRV, Signature, SumType, TypeBase, TypeEnum};
+use crate::Node;
+
+/// Replace the dangling extension pointer in the [`CustomType`]s inside a
+/// signature with a valid pointer to the extension in the `extensions`
+/// registry.
+///
+/// When a pointer is replaced, the extension is added to the
+/// `used_extensions` registry and the new type definition is returned.
+///
+/// This is a helper function used right after deserializing a Hugr.
+pub fn update_op_types_extensions(
```

Comment:
Replaced with inplace `update_op_extensions`.
It doesn't make much of a difference since we are replacing the op entirely, but let's go for consistency.